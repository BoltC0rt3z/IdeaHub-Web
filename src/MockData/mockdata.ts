import { Profile } from '../store/profile/actions/interfaces';
import { Action } from '../store/interfaces';
import { USER_DETAILS } from '../store/profile/actionTypes';
import { SIGN_UP_SUCCESS, SIGN_UP_ERROR } from '../store/signUp/actionTypes';
import { SIGN_UP_STAGES } from '../utils';

export const MockData = {
  Ideas: [...Array(100).keys()].map((item) => ({
    author: { name: 'Brian Ogutu', avatar: 'https://gw.alipayobjects.com/zos/rmsportal/JiqGstEfoWAOHiTxclqi.png' },
    text: `Lorem Ipsum is simply dummy text of the printing and typesetting industry.
		 Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, 
		 when an unknown printer took a galley of type and scrambled it to make a 
		 type specimen book. It has survived not only five centuries, 
		 but also the leap into electronic typesetting, 
		 remaining essentially unchanged. 
		 It was popularised in the
		  1960s with the release of 
		  Letraset sheets containing L
		  orem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.`,
    time: '6:00 PM Â· Apr 2, 2020',
    image: 'https://res.cloudinary.com/authorsheaven/image/upload/v1571160865/57595_1521027489_qlbdng.jpg',
  })),
};

export const testUser: Profile = {
  firstName: 'Brian',
  lastName: 'Ogutu',
  avatar: '',
  username: 'Ogutu-Brian',
};

export const blankTestAction: Action = {
  type: '',
  payload: '',
};

export const blankProfile: Profile = {
  username: '',
  avatar: '',
  firstName: '',
  lastName: '',
};

export const userDetailsAction: Action = {
  type: USER_DETAILS,
  payload: testUser,
};

export const successSignUpAction: Action = {
  type: SIGN_UP_SUCCESS,
  payload: {
    firstName: 'test name',
    lastName: 'test name',
    stage: SIGN_UP_STAGES.VERIFY,
  },
};

export const signUpErrorAction: Action = {
  type: SIGN_UP_ERROR,
  payload: {},
};
export const placeHolderImage: string =
  'https://res.cloudinary.com/authorsheaven/image/upload/v1571160865/57595_1521027489_qlbdng.jpg';

export const profileImage: string =
  'https://res.cloudinary.com/authorsheaven/image/upload/v1589328450/gogo-tomago-hiro-hamada-honey-lemon-tadashi-hamada-animated-film-hero-girl-png-clip-art_afcrtp.png';

export const coverImage: string =
  'https://res.cloudinary.com/authorsheaven/image/upload/v1589369726/93116121_121490372835917_5356212706493857792_o_evnivw.jpg';
