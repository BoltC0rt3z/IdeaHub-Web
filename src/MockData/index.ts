import { MockData } from './mockdata';
export { MockData };