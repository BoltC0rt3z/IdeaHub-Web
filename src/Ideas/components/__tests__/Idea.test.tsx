import React from "react";
import { ReactWrapper, mount } from "enzyme";

import { Idea } from "../";
import { MockData } from '../../../MockData';

describe("Tests for idea", () => {
  const wrapper: ReactWrapper = mount(<Idea idea={MockData.Ideas[0]} />);

  it("Should mount idea without crushing", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
