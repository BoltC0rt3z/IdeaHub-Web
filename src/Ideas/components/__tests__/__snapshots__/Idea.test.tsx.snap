// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for idea Should mount idea without crushing 1`] = `ReactWrapper {}`;
